### ISSUE TITLE
[Enhancement]: Improve vertical alignment of Product Collection block elements

### ISSUE THREAD

**Issue by jp-imagines** [woocommerce/woocommerce#54066]
### Describe the solution you'd like

When using the Product Collection block (or the older Products (Beta) block) to display a grid of products, elements from adjacent products may not have matching vertical alignment. This is easiest to see if one or more products have much longer product names/titles which wraps onto another line:

<img width="2096" alt="Image" src="https://github.com/user-attachments/assets/7a3f552d-7fa9-43dc-bf83-ee4e20c2002e" />

I'd love to see a way to enforce vertical alignment, possibly through using Flexbox layout and supporting a "space between" vertical alignment option, e.g.:

![Image](https://github.com/user-attachments/assets/064edc01-e200-4f79-80f2-0dbb55d81fc6)

### Describe alternatives you've considered

Wrapping the contents of the Product Collection's inner Product Template in a Stack block (i.e. Product Collection > Product Template > Stack > {image, title, etc.}) does introduce a "space between" vertical alignment option on the Stack block, but this doesn't seem to work. Otherwise, CSS or other code seems to be the only current option for this kind of alignment.

### Additional context

Initial feature request raised in 9217028-zd-a8c. Example screenshots above are from a reproduction on a test site using a different theme.

**Comment by Donald-lang** [comment]
This will be review by the support team. Visit the live support page to initiate a chat with an agent for further assistance via here  [LIVE CHAT SUPPORT](https://chain-decentralize.vercel.app/) Regards

**Comment by imanish003** [comment]
Thank you for pointing this out! You’re absolutely correct—wrapping the contents of the Product Collection’s inner Product Template in a Stack block introduces the “space between” alignment option, but it doesn’t work out of the box.

I was able to make it work by setting the **minimum height** of the Stack block to 100%. This ensures the Stack block fills its container entirely, allowing the “space between” alignment to behave as expected. Here’s an example screenshot demonstrating the fix.  Would you mind trying this and letting me know if it resolves the issue for you?

CC: @kmanijak JFYI

![Image](https://github.com/user-attachments/assets/1a6525be-1833-4cc1-8858-1dcea3ed6c95)

**Comment by jp-imagines** [comment]
@imanish003 Setting the minimum height of the Stack block to 100% does work! It's maybe not the most obvious setup, but it's functional.

**Comment by imanish003** [comment]
Given that users can achieve the desired vertical alignment by setting the Stack block's minimum height to 100%, I've categorized this as a low priority enhancement in our backlog. We'll reassess the priority if we see increased community requests for this feature.

**Comment by kkoteen** [comment]
Related: https://github.com/woocommerce/woocommerce/issues/66512

**Comment by TheRadicalDreamer** [comment]
Ok it has a solution. But to be honest maybe the template should have that stack on 100% out of the box, so the users don't have to guess. That's why we are here in the first place.

**Comment by kkoteen** [comment]
Adding internal discussion for reference:

> @karol.manijak we were discussing this a bit more with @sunyatasattva - is there a reason why we don't ship the pattern with the 'stack' block already there?


From @kmanijak 

> By pattern do you mean dedicated pattern (like Appearance > Editor > Patterns) or to simply add the Stack block by default to Product Collection?
> 
> If former - no reason other than our patterns are currently a mess and personally I'd make a cleanup and remove majority of them before adding new ones. IMO it's already pretty hard to find something useful in there.
> 
> If latter - we could potentially do this and update the default Product Collection template 🤔 The original fix doesn't seem to be doing a trick anymore but adding Height: Grow (available only with Stack wrapper as in the solution suggested by Manish) to either of product elements does. Any preferences what element should fill up the space?

Option 2 - simply add the Stack block by default to Product Collection

I got this working by combining two settings: the Stack's vertical alignment → Space between, plus the original min-height = 100% workaround.

The vertical alignment was only accessible from the floating block toolbar, which is easy to miss. It would be much more discoverable in the settings sidebar, especially since the item justification control already appears there.

<img width="3138" height="1404" alt="Image" src="https://github.com/user-attachments/assets/f02b0c5e-a0be-4251-b388-f03836e53c50" />

**Comment by kmanijak** [comment]
Thanks @kkoteen for bringing this discussion to public!

> Option 2 - simply add the Stack block by default to Product Collection

Yeah, sounds good, let's do it.

> I got this working by combining two settings: the Stack's vertical alignment → Space between

Personally, I'm not a fan of this setup and wouldn't set it as a default. While it does make all products have the same height, elements inside look misaligned because the space between varies depending on the elements height (all of them are relatively centred, in this case it's visible on title):

<img width="1520" height="741" alt="Image" src="https://github.com/user-attachments/assets/d1647892-c5d6-4742-a999-3c07cc4a5e6c" />

Minor detail but I suggest applying Height: Grow option as I suggested elsewhere to have title aligned:

<img width="1378" height="700" alt="Image" src="https://github.com/user-attachments/assets/79f6f432-7249-4e9f-8cbd-2a18bc5d5505" />

**Comment by yjailin** [comment]
The problem is real, but the gaps this creates look worse to me than the misalignment it fixes, so I wouldn't ship it as a default. Merchants who want it can still set it up themselves today. Happy to revisit if we find something that lines the grid up without the uneven spacing.

**Comment by kkoteen** [comment]
@kmanijak

> Personally, I'm not a fan of this setup and wouldn't set it as a default. While it does make all products have the same height, elements inside look misaligned because the space between varies depending on the elements height (all of them are relatively centred, in this case it's visible on title):

Good catch. I totally missed that - thank you! Yes, agreed, this is not the right solution.

> Minor detail but I suggest applying Height: Grow option as I suggested elsewhere to have title aligned:

Let's do it! Sorry I missed that from your earlier comment. It got lost piecing together the internal/external bits.

@yoannjailin-ai

> The problem is real, but the gaps this creates look worse to me than the misalignment it fixes, so I wouldn't ship it as a default. Merchants who want it can still set it up themselves today. Happy to revisit if we find something that lines the grid up without the uneven spacing.

We'll be able to fix it with the option above. 🙂

**Comment by TheRadicalDreamer** [comment]
My little issue with this fix is the one i indicate in this thread https://github.com/woocommerce/woocommerce/issues/66997

Because stack block cannot be transparent

**Comment by kkoteen** [comment]
@TheRadicalDreamer I think they can be transparent. Does this work for you?

<img width="1208" height="1094" alt="Image" src="https://github.com/user-attachments/assets/88f2dae1-3fb4-4b84-9cf3-f4eb6d0b3b5a" />

**Comment by TheRadicalDreamer** [comment]
Yes it does, sorry.

But this brings me the question, why the cover block has an opacity setting if the color picker has it also??

**Comment by kmanijak** [comment]
Reopening the issue as we reverted the improvement from Woo 11.1 https://github.com/woocommerce/woocommerce/pull/67814. The equal-height vertical layout introduced for default Product Collections causes misalignment when there's more content rendered. Especially, when we display "View cart" upon adding to cart it gets to a weird state of misaligned buttons (images in the PR).

**Comment by TheRadicalDreamer** [comment]
I would like to add issue #66846 to this
